perception/autoware_tensorrt_yolox/src/tensorrt_yolox_node.cpp

```diff
@@ -128,6 +133,11 @@ void TrtYoloXNode::onConnect()
 
 void TrtYoloXNode::onImage(const sensor_msgs::msg::Image::ConstSharedPtr msg)
 {
+  if (!setCudaDeviceId(gpu_id_)) {
```

Comment:
If possible, I'd also appreciate your consideration whether we surely need to call `setCudaDevicdId` on every iteration. Since `TrtYoloX` class maintains addresses of the GPU memory to avoid memory allocation on every iteration, I think it would be nice for the device ID to be fixed through iterations.

Reply:
I checked this situation, and if I don't put the ```setCudaDeviceId()``` in the callback, I get the following error since the CUDA device switches ID to 0 (it was checked whit ```cudaGetDevice()```).
```
CUDA error: an illegal memory access was encountered
```

Additionally, I checked the previous implementation in the ```tensorrt_yolo``` package, and it also has a very similar function in the callback:

https://github.com/autowarefoundation/autoware.universe/blob/9f1ae6d9e84efa1b3f04ac76d0105853c1237a3b/perception/tensorrt_yolo/src/nodelet.cpp#L131-L140

Reply:
Thanks for the confirmation.  My concern here was additional overhead (i.e., increasing processing time) by introducing setDevice on every iteration. According to the [document](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g159587909ffa0791bbe4b40187a4c6bb), `cudaSetDevice` will take significant time under the limited situation, I guess the practical overhead of this code is not so large.